sdk/iot/Azure.Iot.Hub.Service/src/IoTHubServiceClient.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Iot.Hub.Service
+{
+    /// <summary>
+    /// The IoT Hub Service Client
+    /// </summary>
+    public class IoTHubServiceClient
+    {
+        /// <summary>
+        /// place holder for Devices
+        /// </summary>
+        public DevicesClient Devices;
+        /// <summary>
+        /// place holder for Modules
+        /// </summary>
+        public ModulesClient Modules;
+        /// <summary>
+        /// place holder for Statistics
+        /// </summary>
+        public StatisticsClient Statistics;
+        /// <summary>
+        /// place holder for Messages
+        /// </summary>
+        public CloudToDeviceMessagesClient Messages;
+        /// <summary>
+        /// place holder for Files
+        /// </summary>
+        public FilesClient Files;
+        /// <summary>
+        /// place holder for Jobs
+        /// </summary>
+        public JobsClient Jobs;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
+        /// </summary>
+        public IoTHubServiceClient()
+            : this(new IoTHubServiceClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
+        /// </summary>
+        public IoTHubServiceClient(IoTHubServiceClientOptions options)
+        {
+            Argument.AssertNotNull(options, nameof(options));
+
+            Devices = new DevicesClient();
```

Comment:
I would add a note here explaining that this is where we would pass in the same Http pipeline object (containing the auth credentials) to all of the individual sub-clients.

Reply:
agreed, however this PR is mainly to get started and we can all add notes/comments, write code for http pipeline etc as we start working on this. Or I can add a TBD comments here if you think it is not obvious

Reply:
I think a note here would help, yes.

Reply:
Are we not passing an instance of the client here instead of the http pipeline object?

Reply:
oh yes I recall that. In that case, I'll leave it as it is.

Reply:
That's correct; then we expect all of these sub-clients to be initialized in the `IoTHubServiceClient` constructor:
- the generated sub-client -> with the pipeline object
- our CL sub-clients -> with the generated sub-clients.

Reply:
Sure, let's address this in follow up PRs.